### PR TITLE
Update Blackbird support

### PR DESCRIPTION
**Pull request by stewartsmith**
Pull in Hostboot port 81h/82h progress code
Update Skiboot to report end of IPL
Make blackbird patches Blackbird specific

Signed-off-by: Stewart Smith <stewart@flamingspork.com>

### CODE REVIEW

File: openpower/patches/blackbird-patches/skiboot/0001-platform-add-Raptor-Blackbird-support.patch [hunk 1]
```diff
@@ -92,6 +90,15 @@ index 00000000..8817b8c5
 +	return true;
 +}
 +
++static void blackbird_exit(void)
++{
++	/* Let the IPL Observer know that we're done */
```
Comment (oohal) [inline openpower/patches/blackbird-patches/skiboot/0001-platform-add-Raptor-Blackbird-support.patch]: I thought tpearson was less-than-impressed by having this in skiboot? Could we do this using an init script in petitboot userspace?

**Comment by stewartsmith** [comment]
On Mon, Feb 24, 2020, at 8:33 PM, oohal wrote:
> *@oohal* commented on this pull request.
> 
> In 
> openpower/patches/blackbird-patches/skiboot/0001-platform-add-Raptor-Blackbird-support.patch <https://github.com/open-power/op-build/pull/3471#discussion_r383653695>:
> 
> > @@ -92,6 +90,15 @@ index 00000000..8817b8c5
>  +	return true;
>  +}
>  +
> ++static void blackbird_exit(void)
> ++{
> ++	/* Let the IPL Observer know that we're done */
> I thought tpearson was less-than-impressed by having this in skiboot? 
> Could we do this using an init script in petitboot userspace?

He wants it to really signify that we're "done" IPLing and up at a petitboot UI. Although there's no way for petitboot to signal that to firmware currently, so we should really add a new OPAL call for it? AS is on FSP machines we kinda lie that IPL is done.

Without it, you get no fan control, which is certainly a non-trivial less than ideal share-a-home-office-with-your-partner harmony mode.

**Comment by oohal** [comment]
> He wants it to really signify that we're "done" IPLing and up at a petitboot UI. Although there's no way for petitboot to signal that to firmware currently, so we should really add a new OPAL call for it?

Well, we've already got OPAL calls for reading and writing to the LPC space. The real problem is that they're only accessible to userspace via the debugfs interface at the moment. I suppose we could fix.

> AS is on FSP machines we kinda lie that IPL is done. Without it, you get no fan control, which is certainly a non-trivial less than ideal share-a-home-office-with-your-partner harmony mode.

Hmm, sounds like we need a driver that can signal to the outside world. Possibly via some kind of... operator panel.

**Comment by sharkcz** [comment]
https://github.com/sharkcz/op-build/commit/44def594958d5614a2403ccc55ed94c519c38fe4 does the signaling from skiroot user-space, but it relies on debugfs. Personally I think reporting "IPL done" when exiting skiboot is acceptable until there is a better solution.

**Comment by shenki** [comment]
We do this using an ipmi sensor on all the other BMC platforms. petitboot will signal that the OS has started as the last thing it does before kexec.

**Comment by stewartsmith** [comment]
Superseded by https://github.com/open-power/op-build/pull/3554

**Pull request closed without merging**